app/controllers/media_controller.rb

```diff
@@ -38,8 +38,25 @@ def verify_token
     end
   end
 
+  def auth_check
+    render json: json_for_media_auth
+  end
+
   private
 
+  def json_for_media_auth
+    if can? :read, @media
+      { status: :success }
+    else
+      { status: :must_authenticate,
+        service: {
+          '@id' => iiif_auth_api_url,
+          'label' => 'Stanford-affiliated? Login to view'
+        }
+      }
+    end
+  end
+
```

Comment:
I kinda like having "allowed_params" be the first private method, and rescue_can_can second, but it's not like we adhere to a specific order.  What do you think?   I believe I made that true in the other controllers, FWIW.